### DISCUSSION TITLE
Icecast "Socket is Busy" Error

### DISCUSSION THREAD

**Discussion by Apaullo436** [rtl-airband/RTLSDR-Airband#286] [Q&A]
Hello,

I've setup a few streams, and this is a new error for me.  Everything was config'd at home and worked great (popped the squelch and had no trouble connecting via a DDNS and listening to the static). Deployed the unit onsite - connected to the network, updated the DDNS, launched RTLSDR with good levels...but no stream.mp3.

The error showing up in daemon.log is:

> Could not connect to xxxxxx.duckdns.org:8000/stream.mp3: Socket is busy

Port forwarding is good to go. I've tried a couple other ports, purged and reinstalled everything. Still getting the same error.  Any insight you might have would be really appreciated. Googling "icecast" or "rtlsdr" and "socket is busy" is just a bunch of awkward silence.

Thanks! Cheers.

**Error:**
Oct 30 06:05:25 raspberrypi rtl_airband[5881]: Trying to reconnect to xxxxxx.duckdns.org:8000/stream.mp3...
Oct 30 06:05:25 raspberrypi rtl_airband[5881]: Connecting to xxxxxx.duckdns.org:8000/stream.mp3...
Oct 30 06:05:55 raspberrypi rtl_airband[5881]: Could not connect to xxxxxxx.duckdns.org:8000/stream.mp3: Socket is busy

**rtl_airband.conf:**
devices:
({
type = "rtlsdr";
index = 0;    #RTL-SDR Number
gain = 32;    #Gain Setting
correction = 0;
mode = "scan";
channels:
(
{
freqs = ( 257.8 );
#freqs = ( 257.8, 295.6, 310.8, 227.6, 234.4, 318.8, 340.1 );        #Add Freq's Here
#labels = ( "Ground", "Inner", "Outer", "Terminal", "CLNC Delivery", "Arrival", "Ops" ); #Label Update
#squelch_threshold = -80;
outputs: (
{
type = "icecast";
server = "xxxxxxx.duckdns.org";
port = 8000;   #Port Number Setting
mountpoint = "stream.mp3";
name = "xxxxxxx";   #Name Of Your Choice
genre = "Air Traffic Control";
description = "xxxxxxxxx"; #Description Of Your Choice
username = "source";
password = "xxxxxxx";   #Icecast2 Password
send_scan_freq_tags = false;
}
);
}
);
}
);

**Comment by szpajder** [comment]
It is essentially a timeout. Note that the interval between "Connecting"
and "Could not connect" messages is exactly 30 seconds - this is how long
rtl_airband waits for the shout socket to transition from the "busy" state
to "connected" state. When this timeout expires, the program emits the
"Could not connect" message together with the textual representation of the
current shout socket state (which is "busy" since it's still trying to
connect but rtl_airband does not want to wait any longer).

Do a "telnet xxxxxxx.duckdns.org 8000" from the machine where you run
rtl_airband. It will probably hang on "Connecting..." which probably means
your firewall or port forwarding is messed up. Once you fix this and you
are able to connect with telnet, then rtl_airband will connect as well.

**Reply by Apaullo436** [reply]
Ahh, a timeout makes more sense.  Thanks for steering me in the right direction.

You're correct it won't telnet in to 8000 (connects fine to SSH on 22). Everything seems forwarded properly on the router, https://www.yougetsignal.com/tools/open-ports/ shows it as open, and with the same ports fowarded at home, rtl_airband connects fine. As soon as it's on the remote network it can't get through - even dumped the Pi into the DMZ, fully disabled the Firewall, etc. Nothing. I'm starting to wonder if it's something backend with their ISP.
